text/0027-extensions.md

```diff
@@ -0,0 +1,320 @@
+# Project title
+- Start Date: 2020/08/20
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/27
+
+## Decision, including impact on distributions
+
+We'll use an extension system that supports
+1. discovery—that components know where to "plug themselves in" to the UI
+2. configurability—that implementers can use config files to change which
+    components are plugged in where
+
+We'll build an UI in a new module `@openmrs/esm-implementer-tools`
+for managing that configuration, which will assist implementers
+in changing the config to add, remove, or reorder extensions.
+
+All servers will have a config file `config.json` in the `frontends` directory
+of the application data directory. This file can be written to by the server,
+and managed through the UI.
```

Comment:
@FlorianRappl I was going to have module-spa handle this by providing a REST endpoint for writing that file. How do you think we should handle it in the Docker deployment paradigm?

Reply:
Either via a simple Node.js service (a la import map deployer) or just as a static file. In the end the management of this file should not be our concern.

If we would really love to leverage server-side, then this whole project is wrongly set up. However, we fully focus on making many choices on the UI (i.e., on the client) thus a file like `config.json` should be regarded as a static asset. If not, we should not choose `config.json`, but rather some API from the existing backend to use the DB for storing / modifying the config.

Reply:
Here is the requirement: an implementer should be able to start an OpenMRS instance with Single-SPA out of the box and start configuring the frontend using the implementer tools and be able to persist those changes in a way that's easy to put in VCS.

I'm going to solve that problem for users of openmrs-module-spa. I will be interested in the approach you take for the others.